packages/react-components/react-charts-preview/src/utilities/FocusableTooltipText.tsx

```diff
@@ -1,80 +1,66 @@
+import { Tooltip } from '@fluentui/react-components';
+import { hasOverflow } from './overflow-utils';
 import { getAccessibleDataObject } from './index';
 import { IAccessibilityProps } from '../types/index';
-import { Async } from '../Utilities';
+import { Async } from './async-utils';
 
 interface IFocusableTooltipTextProps {
-  className?: ITooltipHostProps['hostClassName'];
-  content?: ITooltipHostProps['content'];
+  className?: string;
+  content?: string | JSX.Element | JSX.Element[];
   accessibilityData?: IAccessibilityProps;
 }
 
-interface IFocusableTooltipTextState {
-  textOverflow: boolean;
-}
-
-/**
- * Component to make the text focusable when the overflowed content is clipped
- * because of the CSS text-overflow property.
- */
-export class FocusableTooltipText extends React.Component<IFocusableTooltipTextProps, IFocusableTooltipTextState> {
-  private _tooltipChild = React.createRef<HTMLSpanElement>();
-  private _resizeObserver?: ResizeObserver;
-  private _async: Async;
-
-  constructor(props: IFocusableTooltipTextProps) {
-    super(props);
-
-    this.state = {
-      textOverflow: false,
-    };
+export const FocusableTooltipText:  React.FunctionComponent<IFocusableTooltipTextProps> = React.forwardRef<
+HTMLDivElement,
+IFocusableTooltipTextProps
+>((props, forwardedRef) => {
+  const [textOverflow, setTextOverflow] = useState(false);
+  const tooltipChild = useRef<HTMLSpanElement>(null);
+  const async = useRef(new Async()).current;
+  const resizeObserver = useRef<ResizeObserver>();
 
-    this._async = new Async(this);
-  }
-
-  public render(): React.ReactNode {
-    const { className, content, accessibilityData } = this.props;
-
-    return (
-      <TooltipHost overflowMode={TooltipOverflowMode.Self} hostClassName={className} content={content}>
-        <span
-          {...getAccessibleDataObject(accessibilityData)}
-          ref={this._tooltipChild}
-          data-is-focusable={this.state.textOverflow}
-        >
-          {content}
-        </span>
-      </TooltipHost>
-    );
-  }
+  const getTargetElement = useCallback((): HTMLElement | undefined => {
+    if (!tooltipChild.current || !tooltipChild.current.parentElement) {
+      return undefined;
+    }
+    return tooltipChild.current.parentElement;
+  }, [tooltipChild]);
 
-  public componentDidMount(): void {
-    const overflowElement = this._getTargetElement();
-    if (window.ResizeObserver && overflowElement) {
-      this._resizeObserver = new ResizeObserver(this._async.debounce(this._checkTextOverflow, 500));
-      this._resizeObserver.observe(overflowElement);
+  const checkTextOverflow = useCallback(() => {
+    const overflowElement = getTargetElement();
+    const isTextOverflow = !!overflowElement && hasOverflow(overflowElement);
+    if (isTextOverflow !== textOverflow) {
+      setTextOverflow(isTextOverflow);
     }
-  }
+  }, [getTargetElement, textOverflow, setTextOverflow]);
```

Comment:
why is getTargetElement a dependency of this callback

Reply:
overflowElement requires getTargetElement()